File: BraintreeCore/src/main/java/com/braintreepayments/api/IntegrationType.java
package com.braintreepayments.api;

import androidx.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

class IntegrationType {

    @Retention(RetentionPolicy.SOURCE)
    @StringDef({
            IntegrationType.CUSTOM,
            IntegrationType.DROP_IN
    })
    @interface Integration {}
    static final String CUSTOM = "custom";
    static final String DROP_IN = "dropin";
}

File: BraintreeCore/src/main/java/com/braintreepayments/api/IntegrationType.kt
package com.braintreepayments.api

import androidx.annotation.RestrictTo
import androidx.annotation.StringDef
import java.lang.annotation.Retention
import java.lang.annotation.RetentionPolicy

/**
 * @suppress
 */
@RestrictTo(RestrictTo.Scope.LIBRARY_GROUP)
object IntegrationType {
    const val CUSTOM = "custom"
    const val DROP_IN = "dropin"

    @Retention(RetentionPolicy.SOURCE)
    @StringDef(CUSTOM, DROP_IN)
    internal annotation class Integration
}
